src/workers/newView.ts

```diff
@@ -56,56 +56,63 @@ RETURNING "currentStreak", "maxStreak"
 const incrementReadingStreak = async (
   con: DataSource,
   data: DeepPartial<View>,
-): Promise<boolean> => {
-  const users = con.getRepository(User);
-  const repo = con.getRepository(UserStreak);
-  const { userId, timestamp } = data;
+): Promise<boolean> =>
+  con.transaction(async (entityManager) => {
+    const users = entityManager.getRepository(User);
+    const repo = entityManager.getRepository(UserStreak);
+    const { userId, timestamp } = data;
 
-  if (!userId) {
-    return false;
-  }
+    if (!userId) {
+      return false;
+    }
 
-  const viewTime = timestamp ? new Date(timestamp as string) : new Date();
+    const viewTime = timestamp ? new Date(timestamp as string) : new Date();
 
-  // TODO: This code is temporary here for now because we didn't run the migration yet,
-  // so not every user is going to have a row in the user_streak table and we need to create it.
-  // We can get rid of it once/if we implement the migration in https://dailydotdev.atlassian.net/browse/MI-70
-  const user = await users.findOne({
-    where: { id: userId },
-    relations: ['streak'],
-  });
-  const streak = await user?.streak;
-  if (user && !streak) {
-    await repo.save(
-      repo.create({
-        userId,
-        currentStreak: 1,
-        totalStreak: 1,
-        maxStreak: 1,
-        lastViewAt: viewTime,
-      }),
-    );
-  } else {
-    const results = await repo.query(INC_STREAK_QUERY, [
+    // TODO: This code is temporary here for now because we didn't run the migration yet,
+    // so not every user is going to have a row in the user_streak table and we need to create it.
+    // We can get rid of it once/if we implement the migration in https://dailydotdev.atlassian.net/browse/MI-70
```

Comment:
This ticket is already done, does this mean we have implemented the migration? meaning we can get rid of this code?

Reply:
Just mentioned as it was referring to the done ticket.

Reply:
I think the part of the code that this was referring to was for `repo.save` right? Because that was creating the record for old users (the stuff script also needed to create). So we have to make sure the comments is attached properly.